presto-memory/src/main/java/io/prestosql/plugin/memory/MemoryMetadata.java

```diff
@@ -361,7 +364,19 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         }
 
         return Optional.of(new LimitApplicationResult<>(
-                new MemoryTableHandle(table.getId(), OptionalLong.of(limit)),
+                new MemoryTableHandle(table.getId(), OptionalLong.of(limit), OptionalDouble.empty()),
                 true));
     }
+
+    @Override
+    public Optional<ConnectorTableHandle> applySample(ConnectorSession session, ConnectorTableHandle handle, SampleInformation sampleInformation)
+    {
+        MemoryTableHandle table = (MemoryTableHandle) handle;
+
+        if (table.getSampleRatio().isPresent() || sampleInformation.getSampleType() != SYSTEM) {
```

Comment:
If sampling has been pushed down, you can combine with the previous one by multiplying it. If they are equal, then you want to bail out.

Reply:
Also, this is a little tricky. If limit was pushed down first, the sampling has to be over the table+limit. If table sample is pushed first, the limit has to be over table+sample. The logic in MemorySplitManager always does sample before limit.